clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp

```diff
@@ -1533,18 +1582,13 @@ Expected<StringRef> clang(ArrayRef<StringRef> InputFiles, const ArgList &Args,
   };
 
   // Forward all of the `--offload-opt` and similar options to the device.
+  CmdArgs.push_back("-flto");
   for (auto &Arg : Args.filtered(OPT_offload_opt_eq_minus, OPT_mllvm))
     CmdArgs.append(
         {"-Xlinker",
          Args.MakeArgString("--plugin-opt=" + StringRef(Arg->getValue()))});
 
-  if (Triple.isNVPTX() || Triple.isAMDGPU()) {
-    CmdArgs.push_back("-foffload-lto");
-  } else {
-    CmdArgs.push_back("-flto");
-  }
-
-  if (!Triple.isNVPTX() && !Triple.isSPIRV())
```

Comment:
i added this upstream for some work im doing for openmp offloading [here](https://github.com/llvm/llvm-project/commit/cd19f3f787b01481fd687834457686e16fffdbe6#diff-f7dc51acede945c7c8481147e773e9c09230a93b937bee9eab08ee52826cfa8e), if we really need to reomve it for sycl-spirv maybe we need an offloading kind check for sycl?

Reply:
Also a case of unintentional revert, sorry!